File: nvpl_blas/c/srotmg.c
/******************************************************************************
 * Content: 
 *     This example demonstrates use of API as below:
 *     cblas_srotmg
 *
 ******************************************************************************/
#include "example_helper.h"
#include <string.h> // memset

int main() {
    float d1 = 2.3f;
    float d2 = 4.6f;
    float b1 = 1.0f;
    float b2 = 2.0f;
    float P[5];

    printf("\nExample: cblas_srotmg for computing the parameters for a Givens rotation\n");
    printf("\n#### args: d1=%f, d2=%f, b1=%f, b2=%f\n", d1, d2, b1, b2);

    memset(P, 0, 5 * sizeof(float));

    // call cblas_srotmg
    cblas_srotmg(&d1, &d2, &b1, b2, P);

    // print result
    printf("\nThe result of cblas_srotmg:\n");
    printf("d1=%f, d2=%f, b1=%f\n", d1, d2, b1);
    print_svector(P, 5, 1, "P");

    return 0;
}

File: nvpl_blas/f77/srotmg.f
!******************************************************************************
! Content:
!     This example demonstrates use of API as below:
!     SROTMG 
!
!******************************************************************************
      program   SROTMG_MAIN
*
      implicit none

      include "nvpl_blas_f77_blas.fi"

      integer         i
      real            d1, d2, b1, b2, P(5)
      data            d1 /2.3/, d2 /4.6/, b1 /1.0/, b2 /2.0/,
     &                (P(i), i=1,5) /5*0.0/
      

*       External Subroutines
      external        print_svector 
*
*      Executable Statements
*
      print*
      print 99
*
*       Print input data
      print*
      print 100, d1, d2, b1, b2
*
*      Call SROTMG subroutine
      call SROTMG(d1, d2, b1, b2, P)
*
      print*
      print 101
      print 102, d1, d2, b1, b2
      call print_svector(P,5,1,'P')

      stop
 99   format('Example: SROTMG for computing the parameters for a Givens'
     & ' rotation')
 100  format('#### args: d1=',f5.3,', d2=',f5.3,', b1=',f5.3,', '
     &       'b2=',f5.3)
 101  format('The result of SROTMG: ')
 102  format('d1=',f5.3,', d2=',f5.3,', b1=',f5.3,', b2=',f5.3)
      end
